src/main/java/org/swisspush/reststorage/DefaultRedisProvider.java

```diff
@@ -89,13 +90,19 @@ private Future<RedisAPI> connectToRedis() {
         }
 
         if (connecting.compareAndSet(false, true)) {
-            redis = Redis.createClient(vertx, new RedisOptions()
-                    .setConnectionString(createConnectString())
+            RedisOptions redisOptions = new RedisOptions()
                     .setPassword((redisAuth == null ? "" : redisAuth))
                     .setMaxPoolSize(redisMaxPoolSize)
                     .setMaxPoolWaiting(redisMaxPoolWaitingSize)
                     .setPoolRecycleTimeout(redisPoolRecycleTimeoutMs)
-                    .setMaxWaitingHandlers(redisMaxPipelineWaitingSize));
+                    .setMaxWaitingHandlers(redisMaxPipelineWaitingSize);
+            if (configuration.isRedisClustered()) {
+                redisOptions.setType(RedisClientType.CLUSTER);
+                redisOptions.addConnectionString(createConnectString());
+            } else {
+                redisOptions.setConnectionString(createConnectString());
```

Comment:
`redisOptions.setConnectionString(createConnectString());` is the same for the `if`and the `else` case --> move outside of the `if`

Reply:
One is addConnectionString the oder is setConnectionString.